File: src/jsMain/kotlin/com/adoptu/pet-food.kt
package com.adoptu

import kotlinx.browser.document
import kotlinx.dom.*
import org.w3c.dom.*

data class FoodItem(val name: String, val detail: String, val description: String)
data class FoodCategory(val type: String, val title: String, val items: List<FoodItem>)

@JsExport
@JsName("petFood")
object PetFood {
    private val foodData: Map<String, List<FoodCategory>> = mapOf(
        "DOG" to listOf(
            FoodCategory("safe", "Safe Foods", listOf(
                FoodItem("Chicken", "Cooked, boneless", "Lean protein, easy to digest"),
                FoodItem("Rice", "White or brown", "Good source of carbohydrates"),
                FoodItem("Carrots", "Raw or cooked", "Low in calories, high in fiber"),
                FoodItem("Apples", "Without seeds", "Rich in vitamins"),
                FoodItem("Peanut Butter", "Unsalted, no xylitol", "Good protein source")
            )),
            FoodCategory("harmful", "Harmful Foods", listOf(
                FoodItem("Grapes", "All varieties", "Can cause kidney failure"),
                FoodItem("Onions", "All forms", "Damages red blood cells"),
                FoodItem("Chocolate", "All types", "Toxic to dogs"),
                FoodItem("Garlic", "All forms", "Causes anemia")
            )),
            FoodCategory("cannot", "Cannot Eat", listOf(
                FoodItem("Xylitol", "Artificial sweetener", "Highly toxic"),
                FoodItem("Macadamia Nuts", "All forms", "Causes weakness"),
                FoodItem("Avocado", "Pit, skin, all", "Contains persin"),
                FoodItem("Alcohol", "All forms", "Highly toxic")
            ))
        ),
        "CAT" to listOf(
            FoodCategory("safe", "Safe Foods", listOf(
                FoodItem("Cooked Fish", "Without bones", "High in protein"),
                FoodItem("Chicken", "Cooked, plain", "Good protein source"),
                FoodItem("Pumpkin", "Plain, cooked", "Helps digestion"),
                FoodItem("Eggs", "Cooked", "Complete protein")
            )),
            FoodCategory("harmful", "Harmful Foods", listOf(
                FoodItem("Raw Eggs", "With avidin", "Interferes with biotin"),
                FoodItem("Raw Fish", "Contains thiaminase", "Breaks down B vitamins"),
                FoodItem("Dog Food", "Any", "Lacks taurine"),
                FoodItem("Milk", "Most cats", "Lactose intolerance")
            )),
            FoodCategory("cannot", "Cannot Eat", listOf(
                FoodItem("Chocolate", "All types", "Theobromine toxic"),
                FoodItem("Onions/Garlic", "All forms", "Damages RBC"),
                FoodItem("Grapes/Raisins", "All varieties", "Kidney damage"),
                FoodItem("Alcohol", "All forms", "Very toxic")
            ))
        ),
        "BIRD" to listOf(
            FoodCategory("safe", "Safe Foods", listOf(
                FoodItem("Seeds", "Variety mix", "Good fat source"),
                FoodItem("Fruits", "Most varieties", "Vitamins and minerals"),
                FoodItem("Vegetables", "Leafy greens", "Essential nutrients"),
                FoodItem("Pellets", "Quality formulated", "Complete nutrition")
            )),
            FoodCategory("harmful", "Harmful Foods", listOf(
                FoodItem("Avocado", "All parts", "Persin toxic"),
                FoodItem("Fruit Pits", "Apple, peach", "Cyanide traces"),
                FoodItem("Salt", "Any form", "Toxic to birds"),
                FoodItem("Caffeine", "Coffee, tea", "Heart issues")
            )),
            FoodCategory("cannot", "Cannot Eat", listOf(
                FoodItem("Chocolate", "All types", "Theobromine toxic"),
                FoodItem("Onions", "All forms", "Damages RBC"),
                FoodItem("Garlic", "All forms", "Toxic"),
                FoodItem("Mushrooms", "Wild varieties", "Potential toxins")
            ))
        ),
        "FISH" to listOf(
            FoodCategory("safe", "Safe Foods", listOf(
                FoodItem("Flakes", "Quality brands", "Complete fish food"),
                FoodItem("Pellets", "Floating/sinking", "Balanced nutrition"),
                FoodItem("Frozen Food", "Bloodworms, brine", "Protein source"),
                FoodItem("Vegetables", "Blanched zucchini", "Fiber source")
            )),
            FoodCategory("harmful", "Harmful Foods", listOf(
                FoodItem("Bread", "Any type", "No nutrition, fills up"),
                FoodItem("Human Food", "Cooked meals", "Wrong nutrition"),
                FoodItem("Live Feed", "Wild caught insects", "Potential parasites")
            )),
            FoodCategory("cannot", "Cannot Eat", listOf(
                FoodItem("Land Insect", "Crawling bugs", "Not natural food"),
                FoodItem("Mammal Meat", "Chicken, beef", "Wrong for most fish"),
                FoodItem("Dairy", "Any", "Cannot digest"),
                FoodItem("Breadcrumbs", "Many contain garlic", "Harmful additives")
            ))
        ),
        "RABBIT" to listOf(
            FoodCategory("safe", "Safe Foods", listOf(
                FoodItem("Hay", "Timothy grass", "80% of diet"),
                FoodItem("Leafy Greens", "Romaine, cilantro", "Daily vitamins"),
                FoodItem("Pellets", "Timothy-based", "Balanced nutrition"),
                FoodItem("Carrots", "In moderation", "Treat")
            )),
            FoodCategory("harmful", "Harmful Foods", listOf(
                FoodItem("Iceberg Lettuce", "No nutrition", "Can cause GI stasis"),
                FoodItem("Beans", "All legumes", "Gas, digestive issues"),
                FoodItem("Corn", "Any form", "Cannot digest"),
                FoodItem("Potatoes", "Any form", "Wrong nutrients")
            )),
            FoodCategory("cannot", "Cannot Eat", listOf(
                FoodItem("Chocolate", "All types", "Theobromine toxic"),
                FoodItem("Onions", "All forms", "Very toxic"),
                FoodItem("Garlic", "All forms", "Toxic"),
                FoodItem("Avocado", "All parts", "Contains persin")
            ))
        )
    )

    @JsName("showFoodInfo")
    @JvmStatic
    fun showFoodInfo(petType: String) {
        val categories = foodData[petType]
        val container = document.getElementById("food-info")
        if (categories == null || container == null) return
        
        var html = ""
        categories.forEach { cat ->
            html += """<div class="food-category" data-type="${cat.type}"><h3>${cat.title}</h3><ul class="food-list">"""
            cat.items.forEach { item ->
                html += """<li><strong>${item.name}</strong>"""
                if (item.detail.isNotEmpty()) html += """ <span class="food-detail">(${item.detail})</span>"""
                if (item.description.isNotEmpty()) html += """<p class="food-desc">${item.description}</p>"""
                html += "</li>"
            }
            html += "</ul></div>"
        }
        container.innerHTML = html
    }

    private val petTypeNames = mapOf(
        "DOG" to "Dog",
        "CAT" to "Cat",
        "BIRD" to "Bird",
        "FISH" to "Fish",
        "RABBIT" to "Rabbit"
    )

    @JsName("init")
    @JvmStatic
    fun init() {
        val buttons = document.querySelectorAll(".pet-type-btn")
        buttons.forEach { btn: dynamic ->
            btn.addEventListener("click") { 
                document.querySelectorAll(".pet-type-btn").forEach { b: dynamic -> b.classList.remove("active") }
                btn.classList.add("active")
                showFoodInfo(btn.dataset.type)
                updateSelectedPetType(btn.dataset.type)
            }
        }
        showFoodInfo("DOG")
        updateSelectedPetType("DOG")
    }

    private fun updateSelectedPetType(petType: String) {
        val heading = document.getElementById("selected-pet-type")
        if (heading != null) {
            heading.textContent = "${petTypeNames[petType] ?: petType} Food Information"
        }
    }
}

fun init() {
    if (document.readyState == "loading") {
        document.addEventListener("DOMContentLoaded") { PetFood.init() }
    } else {
        PetFood.init()
    }
}

File: backend/src/main/resources/static/js/pet-food.js
const foodData = {
    'DOG': [
        {type:'safe',title:'Safe Foods',items:[
            {name:'Chicken',detail:'Cooked, boneless',description:'Lean protein, easy to digest'},
            {name:'Rice',detail:'White or brown',description:'Good source of carbohydrates'},
            {name:'Carrots',detail:'Raw or cooked',description:'Low in calories, high in fiber'},
            {name:'Apples',detail:'Without seeds',description:'Rich in vitamins'},
            {name:'Peanut Butter',detail:'Unsalted, no xylitol',description:'Good protein source'}
        ]},
        {type:'harmful',title:'Harmful Foods',items:[
            {name:'Grapes',detail:'All varieties',description:'Can cause kidney failure'},
            {name:'Onions',detail:'All forms',description:'Damages red blood cells'},
            {name:'Chocolate',detail:'All types',description:'Toxic to dogs'},
            {name:'Garlic',detail:'All forms',description:'Causes anemia'}
        ]},
        {type:'cannot',title:'Cannot Eat',items:[
            {name:'Xylitol',detail:'Artificial sweetener',description:'Highly toxic'},
            {name:'Macadamia Nuts',detail:'All forms',description:'Causes weakness'},
            {name:'Avocado',detail:'Pit, skin, all',description:'Contains persin'},
            {name:'Alcohol',detail:'All forms',description:'Highly toxic'}
        ]}
    ],
    'CAT': [
        {type:'safe',title:'Safe Foods',items:[
            {name:'Cooked Fish',detail:'Without bones',description:'High in protein'},
            {name:'Chicken',detail:'Cooked, plain',description:'Good protein source'},
            {name:'Pumpkin',detail:'Plain, cooked',description:'Helps digestion'},
            {name:'Eggs',detail:'Cooked',description:'Complete protein'}
        ]},
        {type:'harmful',title:'Harmful Foods',items:[
            {name:'Raw Eggs',detail:'With avidin',description:'Interferes with biotin'},
            {name:'Raw Fish',detail:'Contains thiaminase',description:'Breaks down B vitamins'},
            {name:'Dog Food',detail:'Any',description:'Lacks taurine'},
            {name:'Milk',detail:'Most cats',description:'Lactose intolerance'}
        ]},
        {type:'cannot',title:'Cannot Eat',items:[
            {name:'Chocolate',detail:'All types',description:'Theobromine toxic'},
            {name:'Onions/Garlic',detail:'All forms',description:'Damages RBC'},
            {name:'Grapes/Raisins',detail:'All varieties',description:'Kidney damage'},
            {name:'Alcohol',detail:'All forms',description:'Very toxic'}
        ]}
    ],
    'BIRD': [
        {type:'safe',title:'Safe Foods',items:[
            {name:'Seeds',detail:'Variety mix',description:'Good fat source'},
            {name:'Fruits',detail:'Most varieties',description:'Vitamins and minerals'},
            {name:'Vegetables',detail:'Leafy greens',description:'Essential nutrients'},
            {name:'Pellets',detail:'Quality formulated',description:'Complete nutrition'}
        ]},
        {type:'harmful',title:'Harmful Foods',items:[
            {name:'Avocado',detail:'All parts',description:'Persin toxic'},
            {name:'Fruit Pits',detail:'Apple, peach',description:'Cyanide traces'},
            {name:'Salt',detail:'Any form',description:'Toxic to birds'},
            {name:'Caffeine',detail:'Coffee, tea',description:'Heart issues'}
        ]},
        {type:'cannot',title:'Cannot Eat',items:[
            {name:'Chocolate',detail:'All types',description:'Theobromine toxic'},
            {name:'Onions',detail:'All forms',description:'Damages RBC'},
            {name:'Garlic',detail:'All forms',description:'Toxic'},
            {name:'Mushrooms',detail:'Wild varieties',description:'Potential toxins'}
        ]}
    ],
    'FISH': [
        {type:'safe',title:'Safe Foods',items:[
            {name:'Flakes',detail:'Quality brands',description:'Complete fish food'},
            {name:'Pellets',detail:'Floating/sinking',description:'Balanced nutrition'},
            {name:'Frozen Food',detail:'Bloodworms, brine',description:'Protein source'},
            {name:'Vegetables',detail:'Blanched zucchini',description:'Fiber source'}
        ]},
        {type:'harmful',title:'Harmful Foods',items:[
            {name:'Bread',detail:'Any type',description:'No nutrition, fills up'},
            {name:'Human Food',detail:'Cooked meals',description:'Wrong nutrition'},
            {name:'Live Feed',detail:'Wild caught insects',description:'Potential parasites'}
        ]},
        {type:'cannot',title:'Cannot Eat',items:[
            {name:'Land Insect',detail:'Crawling bugs',description:'Not natural food'},
            {name:'Mammal Meat',detail:'Chicken, beef',description:'Wrong for most fish'},
            {name:'Dairy',detail:'Any',description:'Cannot digest'},
            {name:'Breadcrumbs',detail:'Many contain garlic',description:'Harmful additives'}
        ]}
    ],
    'RABBIT': [
        {type:'safe',title:'Safe Foods',items:[
            {name:'Hay',detail:'Timothy grass',description:'80% of diet'},
            {name:'Leafy Greens',detail:'Romaine, cilantro',description:'Daily vitamins'},
            {name:'Pellets',detail:'Timothy-based',description:'Balanced nutrition'},
            {name:'Carrots',detail:'In moderation',description:'Treat'}
        ]},
        {type:'harmful',title:'Harmful Foods',items:[
            {name:'Iceberg Lettuce',detail:'No nutrition',description:'Can cause GI stasis'},
            {name:'Beans',detail:'All legumes',description:'Gas, digestive issues'},
            {name:'Corn',detail:'Any form',description:'Cannot digest'},
            {name:'Potatoes',detail:'Any form',description:'Wrong nutrients'}
        ]},
        {type:'cannot',title:'Cannot Eat',items:[
            {name:'Chocolate',detail:'All types',description:'Theobromine toxic'},
            {name:'Onions',detail:'All forms',description:'Very toxic'},
            {name:'Garlic',detail:'All forms',description:'Toxic'},
            {name:'Avocado',detail:'All parts',description:'Contains persin'}
        ]}
    ]
};

function showFoodInfo(petType) {
    const data = foodData[petType];
    const container = document.getElementById('food-info');
    if (!data || !container) return;
    let html = '';
    data.forEach(function(cat) {
        html += '<div class="food-category"><h3>' + cat.title + '</h3><ul class="food-list">';
        cat.items.forEach(function(i) {
            html += '<li><strong>' + i.name + '</strong>';
            if (i.detail) html += ' <span class="food-detail">(' + i.detail + ')</span>';
            if (i.description) html += '<p class="food-desc">' + i.description + '</p>';
            html += '</li>';
        });
        html += '</ul></div>';
    });
    container.innerHTML = html;
}

document.querySelectorAll('.pet-type-btn').forEach(function(btn) {
    btn.addEventListener('click', function() {
        document.querySelectorAll('.pet-type-btn').forEach(function(b) { b.classList.remove('active'); });
        this.classList.add('active');
        showFoodInfo(this.dataset.type);
    });
});

showFoodInfo('DOG');